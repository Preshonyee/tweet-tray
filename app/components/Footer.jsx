import React from 'react';
import PropTypes from 'prop-types';

import Styled from 'styled-components';

import * as constants from '../constants';

const FooterStyle = Styled.footer`
  position: absolute;
  z-index: 99;
  left: 0px;
  bottom: 0px;
  width: 100%;
  height: 38px;
  padding-top: 8px;
  padding-bottom: 8px;
  background-color: ${(props) => {
    return props.theme === 'day' ? constants.WHITE : constants.DARK_MODE_FOREGROUND;
  }};
  border-top: 1px solid ${(props) => {
    return props.theme === 'day' ? constants.BORDER_GREY : constants.DARK_MODE_BACKGROUND;
  }};
`;

const LeftViewStyle = Styled.div`
  max-width: 50%;
  height: 100%;
  float: left;
  padding-left: ${constants.SPACING}px;

  & > * {
    float: left;
  }
`;

const RightViewStyle = Styled.div`
  max-width: 50%;
  height: 100%;
  float: right;
  padding-right: ${constants.SPACING}px;

  & > * {
    float: right;
  }
`;

const Footer = (props) => {
  const { leftView, rightView, theme, } = props;
  return (
    <FooterStyle
      theme={theme}
    >
      {leftView !== null && (
        <LeftViewStyle>
          {leftView}
        </LeftViewStyle>
      )}
      {rightView !== null && (
        <RightViewStyle>
          {rightView}
        </RightViewStyle>
      )}
    </FooterStyle>
  );
};

Footer.propTypes = {
  theme: PropTypes.string,
  leftView: PropTypes.object,
  rightView: PropTypes.object,
};

Footer.defaultProps = {
  theme: 'day',
  leftView: null,
  rightView: null,
};

export default Footer;
