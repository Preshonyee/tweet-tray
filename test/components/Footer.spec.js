import React from 'react';
import Enzyme, { shallow, } from 'enzyme';
import { shallowToJson, } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Footer from '../../app/components/Footer';

Enzyme.configure({ adapter: new Adapter(), });

function setup() {
  const component = shallow(<Footer
    leftView={<div>Left View</div>}
    rightView={<div>Right View</div>}
  />);
  return {
    component,
    leftView: component.find('Footer__LeftViewStyle').find('div'),
    rightView: component.find('Footer__RightViewStyle').find('div'),
  };
}

describe('Footer component', () => {
  it('should render', () => {
    const { component, } = setup();
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should render with leftView content', () => {
    const { leftView, } = setup();
    expect(shallowToJson(leftView)).toMatchSnapshot();
  });

  it('should render with rightView content', () => {
    const { rightView, } = setup();
    expect(shallowToJson(rightView)).toMatchSnapshot();
  });
});
