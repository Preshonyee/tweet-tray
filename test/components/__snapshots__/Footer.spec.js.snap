// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should render 1`] = `
<Footer__FooterStyle
  theme="day"
>
  <Footer__LeftViewStyle>
    <div>
      Left View
    </div>
  </Footer__LeftViewStyle>
  <Footer__RightViewStyle>
    <div>
      Right View
    </div>
  </Footer__RightViewStyle>
</Footer__FooterStyle>
`;

exports[`Footer component should render with leftView content 1`] = `
<div>
  Left View
</div>
`;

exports[`Footer component should render with rightView content 1`] = `
<div>
  Right View
</div>
`;
